File: Learning RAC/Learning RAC/ViewController.m
//
//  ViewController.m
//  Learning RAC
//
//  Created by Wes Van der Klomp on 6/20/16.
//  Copyright © 2016 wvdk. All rights reserved.
//

#import "ViewController.h"
#import <ReactiveCocoa/ReactiveCocoa.h>

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIButton *greenButton;
@property (weak, nonatomic) IBOutlet UIButton *blueButton;
@property (weak, nonatomic) IBOutlet UIButton *redButton;
@property (weak, nonatomic) IBOutlet UIView *colorBox;
@property (weak, nonatomic) IBOutlet UIButton *killGreenButtonButton;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    ///////////////////////////////////////////////
    //: ## Making my first signal
    
    RACSignal *letters = [@"A B C D E F G H I" componentsSeparatedByString:@" "].rac_sequence.signal;
    
    [letters subscribeNext:^(NSString *x) {
        NSLog(@"Letters: %@", x);
    }];
    
    ///////////////////////////////////////////////
    //: ## Three ways to make a signal for a button
    
    //: ### Red button
    
    RACSignal *makeRed = [_redButton rac_signalForControlEvents:UIControlEventTouchUpInside];
    
    [makeRed subscribeNext:^(id sender) {
        NSLog(@"Red button was pressed!");
        _colorBox.backgroundColor = UIColor.redColor;
    }];
    
    //: ### Blue button
    
    [[_blueButton rac_signalForControlEvents:UIControlEventTouchDown] subscribeNext:^(id sender) {
        NSLog(@"Blue button was pressed!");
        _colorBox.backgroundColor = UIColor.blueColor;
    }];
    
    //: ### Green button

    void (^makeBoxGreen)(id) = ^void(id x) {
        NSLog(@"Green, man. Green.");
        _colorBox.backgroundColor = UIColor.greenColor;
    };
    
    [[_greenButton rac_signalForControlEvents:UIControlEventTouchDown] subscribeNext:makeBoxGreen completed:^(void){
        NSLog(@"That's all folks. No more green for you.");
    }];

    ///////////////////////////////////////////////
    //: ## Lets try breaking things and see what happens
    

    
//    [_killGreenButtonButton rac_signalForControlEvents:UIContProlEventTouchDown] subscribe:<#(id<RACSubscriber>)#>
    
    
    
    
    
    
}

@end

File: Learning Swift and Cocoa/really old/On Off 3/On Off 3/ViewController.swift
//
//  ViewController.swift
//  On Off 3
//
//  Created by Wes Van der Klomp on 4/16/15.
//  Copyright (c) 2015 Wes Van der Klomp. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet var bg: UIView!
    @IBOutlet weak var switchOutlet: UISwitch!
    
    @IBAction func switched(sender: AnyObject) {
        if bg.backgroundColor != UIColor.blueColor() {
            bg.backgroundColor = UIColor.blueColor()
        } else {
            bg.backgroundColor = UIColor.redColor()
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        bg.backgroundColor = UIColor.redColor()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

